src/main/thrift/parquet.thrift

```diff
@@ -384,6 +384,11 @@ struct BsonType {
  * Embedded Variant logical type annotation
  */
 struct VariantType {
+  // If the Variant is shredded the version of the shredding specification used.
+  // Required, if the the column is shredded.
+  //
+  // See VariantShredding.md for differences between versions.
+  1: optional i8 shredding_version
```

Comment:
What about encoding version? If we include the encoding version, that would help us change it in the future. For shredding, do we want to have a single version that includes shredding or a separate version for shredding?

Reply:
encoding version is already embedded in metadata, so I thought wouldn't be needed here. Shredding version is only encoded implicitly based on the spec without this.  

I think shredding should probably evolve differently from encoding (hopefully we won't need to evolve any for some time).  But shredding seems like we need at least some more experimentation to determine if there are other forms that might be better.  

I might not be fully understanding the question here though.

Reply:
Would that mean that we allow mixed versions based on the metadata for each record? That seems like unnecessary complication to me. And it also means that older clients would fail at read time when they encounter a newer record, rather than failing quickly at the schema check stage.

I think it makes sense to put the encoding version here and expect uniform encoding throughout a Parquet file. Writers should produce the latest encoding, not carry records through.

For the shredding question, I would rather have one version of variant instead of evolving them separately. I think that would get confusing and there would be dependencies between them. For example, can you shred a type defined by a newer version of the encoding? Probably not, so bumping the encoding version also requires bumping the shredding version. So is it worth it to have one version number that can increase independently? I would combine them into a single variant version for shredding and encoding.